code/Extensions/SubmittedFileFieldExtension.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace SilverStripe\UserForms\Extensions;
+
+use SilverStripe\Assets\File;
+use SilverStripe\Core\Extension;
+use SilverStripe\Security\Security;
+
+class SubmittedFileFieldExtension extends Extension
```

Comment:
That's not what I had in mind.

If there's an easy way to grant `canView` access to a file based on whether it's attached to a user form the member has access to, I think we could add that to the changelog. However, I don't we shouldn't support or facilitate this use case out of the box.

On the specific way this could be achieved, there's a `UserFormFileExtension` that gets attached to `File` by the userform module. This creates a `SubmittedFileField` "belongs to" relation.

https://github.com/silverstripe/silverstripe-userforms/blob/4ea877c7e3064564a762d55ea14a1490b9e69373/code/Extension/UserFormFileExtension.php#L33

In theory, a user could create an extension like this to tweak the canView permissions.

```php
class SubmittedFileFieldExtension extends DataExtension
{
    public function canView(Member $member)
    {
        $submittedFileField = $this->getOwner()->SubmittedFileField;
        if ($submittedFileField && $submittedFileField->canView($member)) {
            return true;
        }
    }
}
```

I haven't tested this, so I don't know for a fact this will work.